### ISSUE TITLE
vmware-iso: unexpected EOF

### ISSUE THREAD

**Issue by johnjelinek** [hashicorp/packer#5429]
I'm running Packer 1.1.0 on macOS connecting to a remove vware esx6 host:

```
vmware-iso output will be in this color.

==> vmware-iso: Downloading or copying ISO
    vmware-iso: Downloading or copying: http://mirrors.kernel.org/centos/7.4.1708/isos/x86_64/CentOS-7-x86_64-DVD-1708.iso
==> vmware-iso: Uploading ISO to remote machine...
==> vmware-iso: Creating virtual machine disk
==> vmware-iso: Building and writing VMX file
==> vmware-iso: Starting HTTP server on port 8798
==> vmware-iso: Registering remote VM...
==> vmware-iso: Starting virtual machine...
==> vmware-iso: Waiting 10s for boot...
==> vmware-iso: Connecting to VM via VNC
==> vmware-iso: Stopping virtual machine...
==> vmware-iso: Unregistering virtual machine...
Build 'vmware-iso' errored: unexpected EOF

==> Some builds didn't complete successfully and had errors:
--> vmware-iso: unexpected EOF

==> Builds finished but no artifacts were created.
panic: runtime error: invalid memory address or nil pointer dereference
2017/10/05 17:49:08 packer: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1f77ec5]
2017/10/05 17:49:08 packer:
2017/10/05 17:49:08 packer: goroutine 223 [running]:
2017/10/05 17:49:08 packer: github.com/hashicorp/packer/builder/vmware/iso.(*ESX5Driver).HostIP(0xc420506160, 0x0, 0x0, 0x0, 0x0)
2017/10/05 17:49:08 packer:     /private/tmp/packer-20170913-40656-n4jrpj/src/github.com/hashicorp/packer/builder/vmware/iso/driver_esx5.go:168 +0x175
2017/10/05 17:49:08 packer: github.com/hashicorp/packer/builder/vmware/common.(*StepTypeBootCommand).Run(0xc420492080, 0x3cd1a40, 0xc420444090, 0x0)
2017/10/05 17:49:08 packer:     /private/tmp/packer-20170913-40656-n4jrpj/src/github.com/hashicorp/packer/builder/vmware/common/step_type_boot_command.go:98 +0x8da
2017/10/05 17:49:08 packer: github.com/hashicorp/packer/common.askStep.Run(0x3ccb380, 0xc420492080, 0x3cd4a60, 0xc420414100, 0x3cd1a40, 0xc420444090, 0xc420444098)
2017/10/05 17:49:08 packer:     /private/tmp/packer-20170913-40656-n4jrpj/src/github.com/hashicorp/packer/common/multistep_runner.go:95 +0x49
2017/10/05 17:49:08 packer: github.com/hashicorp/packer/common.(*askStep).Run(0xc4200e4ba0, 0x3cd1a40, 0xc420444090, 0x0)
2017/10/05 17:49:08 packer:     <autogenerated>:1 +0x6b
2017/10/05 17:49:08 packer: github.com/hashicorp/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc4204820c0, 0x3cd1a40, 0xc420444090)
2017/10/05 17:49:08 packer:     /private/tmp/packer-20170913-40656-n4jrpj/src/github.com/hashicorp/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x1ee
2017/10/05 17:49:08 packer: github.com/hashicorp/packer/builder/vmware/iso.(*Builder).Run(0xc4203d8000, 0x3cd4a60, 0xc420414100, 0x3ccbcc0, 0xc4204f6330, 0x3cd33c0, 0xc420166008, 0x80000000018, 0x28, 0x28, ...)
2017/10/05 17:49:08 packer:     /private/tmp/packer-20170913-40656-n4jrpj/src/github.com/hashicorp/packer/builder/vmware/iso/builder.go:319 +0x12bc
2017/10/05 17:49:08 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc4203e4340, 0x1, 0xc420522e70, 0x0, 0x0)
2017/10/05 17:49:08 packer:     /private/tmp/packer-20170913-40656-n4jrpj/src/github.com/hashicorp/packer/packer/rpc/builder.go:93 +0x1dd
2017/10/05 17:49:08 packer: reflect.Value.call(0xc42017f020, 0xc420166230, 0x13, 0x2c39beb, 0x4, 0xc420427f20, 0x3, 0x3, 0x0, 0x0, ...)
2017/10/05 17:49:08 packer:     /usr/local/Cellar/go/1.9/libexec/src/reflect/value.go:434 +0x906
2017/10/05 17:49:08 packer: reflect.Value.Call(0xc42017f020, 0xc420166230, 0x13, 0xc4203be720, 0x3, 0x3, 0x0, 0x0, 0x0)
2017/10/05 17:49:08 packer:     /usr/local/Cellar/go/1.9/libexec/src/reflect/value.go:302 +0xa4
2017/10/05 17:49:08 packer: net/rpc.(*service).call(0xc42015ef80, 0xc420160c30, 0xc42016e178, 0xc420168d80, 0xc4201582c0, 0x26b9400, 0xc420522e6c, 0x18a, 0x26606c0, 0xc420522e70, ...)
2017/10/05 17:49:08 packer:     /usr/local/Cellar/go/1.9/libexec/src/net/rpc/server.go:381 +0x142
2017/10/05 17:49:08 packer: created by net/rpc.(*Server).ServeCodec
2017/10/05 17:49:08 packer:     /usr/local/Cellar/go/1.9/libexec/src/net/rpc/server.go:475 +0x36b
2017/10/05 17:49:08 [INFO] (telemetry) ending vmware-iso
2017/10/05 17:49:08 [INFO] (telemetry) found error: unexpected EOF
2017/10/05 17:49:08 ui error: Build 'vmware-iso' errored: unexpected EOF
2017/10/05 17:49:08 Builds completed. Waiting on interrupt barrier...
2017/10/05 17:49:08 machine readable: error-count []string{"1"}
2017/10/05 17:49:08 ui error:
==> Some builds didn't complete successfully and had errors:
2017/10/05 17:49:08 machine readable: vmware-iso,error []string{"unexpected EOF"}
2017/10/05 17:49:08 ui error: --> vmware-iso: unexpected EOF
2017/10/05 17:49:08 ui:
==> Builds finished but no artifacts were created.
2017/10/05 17:49:08 /usr/local/bin/packer: plugin process exited
2017/10/05 17:49:08 waiting for all plugin processes to complete...
2017/10/05 17:49:08 /usr/local/bin/packer: plugin process exited
2017/10/05 17:49:08 /usr/local/bin/packer: plugin process exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/hashicorp/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

**Comment by mwhooker** [comment]
thanks for the report!

**Comment by johnjelinek** [comment]
@mwhooker: let me know if you need that `crash.log`.

**Comment by mwhooker** [comment]
fixed in 2a326b517

If you're able to reproduce the issue, it would be great if you could test master. Thanks again

**Comment by tonymi** [comment]
@mwhooker 
Hi, I got the same error message, but it happened while waiting for WinRM connection, and Packer didn't clean up the VM. Does the same fix apply? If it does, what is the Packer release version that has the fix?
